src/ServiceControl.Plugin.Nsb6.SagaAudit/CaptureSagaStateBehavior.cs

```diff
@@ -44,36 +43,40 @@ public void Invoke(IncomingContext context, Action next)
             }
 
             sagaAudit.FinishTime = DateTime.UtcNow;
-            AuditSaga(saga, context);
+            await AuditSaga(saga, context).ConfigureAwait(false);
         }
 
-        void AuditSaga(Saga saga, IncomingContext context)
+        Task AuditSaga(Saga saga, IInvokeHandlerContext context)
         {
             string messageId;
 
-            if (!context.IncomingLogicalMessage.Headers.TryGetValue(Headers.MessageId, out messageId))
+            if (!context.MessageHeaders.TryGetValue(Headers.MessageId, out messageId))
             {
-                return;
+                return Task.FromResult(0);
             }
 
-            var activeSagaInstance = context.Get<ActiveSagaInstance>();
-            var sagaStateString = Serializer.Serialize(saga.Entity);
-            var messageType = context.IncomingLogicalMessage.MessageType.FullName;
-            var headers = context.IncomingLogicalMessage.Headers;
+            var activeSagaInstance = context.Extensions.Get<ActiveSagaInstance>();
+
+            // TODO: HOW?? Internalize Json??
+            var sagaStateString = string.Empty;
```

Comment:
@Particular/servicecontrol-maintainers There is probably no way around internalizing JSON.NET for SagaAudits since we cannot longer use the JSonSerializers of the core. If the answer to internalizing is yes, then I'd also switch the sending of the raw message to the internalized JSonSerializer instead of the DataContract one. This would be different then to CustomChecks and Heartbeats // @Particular/nservicebus-maintainers